config/types/timeouts.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"encoding/json"
+)
+
+const (
+	defaultResponseHeaders int = 10
+	defaultTotal           int = 0
+)
+
+type Timeouts struct {
+	HttpResponseHeaders int `json:"httpResponseHeaders,omitempty"`
+	HttpTotal           int `json:"httpTotal,omitempty"`
+}
+
+type timeouts Timeouts
+
+func (t *Timeouts) UnmarshalJSON(data []byte) error {
+	t.HttpResponseHeaders = defaultResponseHeaders
+	t.HttpTotal = defaultTotal
+	th := timeouts(*t)
```

Comment:
> So it doesn't apply defaults if it exits due to error in json.UnmarshalJSON()?

Correct.

> In this case don't we want the defaults anyway?

No. We shouldn't modify the underlying object unless unmarshalling was successful.

> should the other types that work this way (e.g. filesystems, raid, etc) be changed as well?

Those types work the way I described. They don't need to change.